drivers/serial/Kconfig.stm32

```diff
@@ -9,6 +9,7 @@ config UART_STM32
 	depends on DT_HAS_ST_STM32_UART_ENABLED
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	select SERIAL_SUPPORT_DATA_LINE_INVERSION
```

Comment:
Not actually the case on some STM32 series

Reply:
At hardware level i think this means LL_USART_[TX/RX]PIN_LEVEL_INVERTED. but at driver level these properties was included by default in struct uart_stm32_config.
thus keeping it enabled at driver level as it was.